### ISSUE TITLE
Add the `Repository` message type to `Group`, `Enrollment`.

### ISSUE THREAD

**Issue by JosteinLindhom** [quickfeed/quickfeed#972]
Currently we use the `GetRepositories` RPC to fetch all repository types, along with their URL. The URL for the user repository returned is specific to the user invoking the method.

In teacher views we may want to list all enrollments or groups for a course, along with a link to their repository. We have no way of obtaining the URL for enrolled users or groups. What we currently do is to use helper methods to build the links:

```ts
export const userLink = (user: User): string => {
    return `https://github.com/${user.Login}`
}

export const userRepoLink = (course: Course, user: User): string => {
    return `https://github.com/${course.ScmOrganizationName}/${user.Login}-labs`
}

export const groupRepoLink = (course: Course, group: Group): string => {
    return `https://github.com/${course.ScmOrganizationName}/${group.name}`
}
```

We should consider adding the `Repository` type to `Group` and `Enrollment`. Doing so we can get rid of the helpers, and we do not need to worry about updating the logic if any changes to repository naming are introduced.

A user may have multiple repositories. We can use the composite key `(ScmOrganizationID, userID)` (or `(ScmOrganizationID, groupID)`) to fetch repositories for a given course. These can be preloaded into both enrollments and groups.

**Comment by JoachimTislov** [comment]
Refer: #1098, Are you using `GetRepositories` in the frontend and is this still a relevant issue, @JosteinLindhom ?

**Comment by JosteinLindhom** [comment]
> Refer: [#1098](https://github.com/quickfeed/quickfeed/issues/1098), Are you using `GetRepositories` in the frontend and is this still a relevant issue, [@JosteinLindhom](https://github.com/JosteinLindhom) ?

Yes we use the `GetRepositories` rpc method. Although we can likely remove the rpc method if we add the repositories to `Group` and `Enrollment`.

**Comment by JoachimTislov** [comment]
@JosteinLindhom Ok I understand. I can have a look at this
